compiler/rustc_arena/src/lib.rs

```diff
@@ -114,6 +117,102 @@ impl<T> Default for TypedArena<T> {
     }
 }
 
+#[inline]
+unsafe fn write_from_iter<'a, T, I: Iterator<Item = T>>(
```

Comment:
Can we add a comment on the unsafe here, i.e., what are the preconditions necessary?

Reply:
Actually, extracting this function is not necessary because of the upcoming changes prompted by https://github.com/rust-lang/rust/pull/78569#discussion_r526506964